### ISSUE TITLE
Vibration strenght is always at 100% in Metal Gear Solid

### ISSUE THREAD

**Issue by DamianPereira** [libretro/swanstation#72]
Hi! In the options menu when you boot metal gear solid with a dual-shock controller, there's a vibration test option, which has weak and strong test options. On Duckstation the strong and weak vibration are clearly different. In the config there's even a large motor and small motor mappings from SDL.

Is this something that can be fixed on the swanstation end? Or would it require for libretro to support sdl.

**Comment by TideGear** [comment]
Note! The above only fixes the vibration in Bluetooth mode! Working on a USB fix too.

**Comment by xTMODx** [comment]
Hi @TideGear I want vibration on Android for a long time and I thought it is not possible atm, how can you have vibration, what is your device? Maybe you can help me get vibration on my Bluetooth controller too?

**Comment by TideGear** [comment]
Hi @xTMODx. What controller do you have? Did you try the latest RetroArch build?

**Comment by xTMODx** [comment]
I have Flydigi Apex 5 build from today and android 16 on galaxy s25 ultra

**Comment by TideGear** [comment]
@xTMODx Is it in XInput mode and connected via Bluetooth?

**Comment by xTMODx** [comment]
yes correct, it is in xinput mode and connected via bluetooth, I have rumble in standalone duckstation, m64 plus and some others, so that should work, the funny thing is, I have tested RA windows version with Winlator emulator on android and had vibration on PSX and N64 emulators, performance was good too but that is not how I want to use RA on android

**Comment by TideGear** [comment]
![Image](https://github.com/user-attachments/assets/bf5f3d2e-420a-4997-8e1a-70ba1808b06b)

Can you show me what your standalone Android DuckStation has here?

**Comment by xTMODx** [comment]
<img width="1440" height="3120" alt="Image" src="https://github.com/user-attachments/assets/8a50ecd6-1bbd-4da8-a697-281c15c0a3d5" />

**Comment by TideGear** [comment]
![Image](https://github.com/user-attachments/assets/5ee1fef3-06b5-430b-a982-56ae0983f1cb)

Before I dig deeper, do you have your input set up like this in RA on Android? If not, set it and restart RA.

**Comment by xTMODx** [comment]
<img width="3120" height="1440" alt="Image" src="https://github.com/user-attachments/assets/1347cf96-bcd8-448c-ab40-4988db6c2856" />

Edit: the phone vibrate but the controller doesnt.

**Comment by TideGear** [comment]
You've got no vibration at all or just partial?

**Comment by xTMODx** [comment]
The phone does vibrate as it should but no vibration at all on the controller.
Found an older issue here https://github.com/libretro/RetroArch/issues/10338

**Comment by TideGear** [comment]
Using the latest _nightly_ build?

**Comment by xTMODx** [comment]
yes, downloaded two times today to test

**Comment by TideGear** [comment]
Try disabling “Enable Device Vibration”.

**Comment by xTMODx** [comment]
omg yes it worked, I did tested it sometimes in the past but not today, wow thank you for taking your time and help me

**Comment by TideGear** [comment]
Glad to hear it!

**Comment by xTMODx** [comment]
Does it work only for PSX cores atm or all cores? I am trying to get rumble working with N64 right now but no luck so far
Edit: looks like it is working with N64 cores too, awesome!

**Comment by TideGear** [comment]
Is your Player 1 pak set to rumble?

**Comment by xTMODx** [comment]
Yes, it didn't worked first with Mupen64Plus-Next GLES 3 then I tested it with ParaLLEI N64 and had rumble, then I changed RSP plugin in Mupen64Plus and had rumble too, could not reproduce the not working state anymore, so I don't know what fixed it

**Comment by TideGear** [comment]
Weird! Glad you got it working though!
